lib/kuber_hex/gateways/authorize_net.ex

```diff
@@ -33,68 +59,222 @@ defmodule Gringotts.Gateways.AuthorizeNet do
+  @spec refund(Float, String.t, List) :: Tuple
   def refund(amount, id, opts) do
     request_data = normal_refund(amount, id, opts, @transaction_type[:refund])
     response_data = commit(:post, request_data)
-    respond(@response_type[:transaction_response], response_data)
+    respond(response_data)
   end
 
   @doc """
-    Use this method to cancel either an original transaction that is not settled or 
-    an entire order composed of more than one transaction. It can be submitted against
-    any other transaction type. Requires the transId of a request passed as id.
+  To void a transaction
+
+  Use this method to cancel either an original transaction that is not settled or 
+  an entire order composed of more than one transaction. It can be submitted against
+  any other transaction type. Requires the transId of a request passed as id.
+  ## Required Fields
+      opts = [config: %{name:" ", transactionKey: " "}]
+  ## Optional fields
+      opts = [refId: ]
+
+  ## Example
+      opts = [config: %{name: "your_name", transactionKey: "your_key"},
+        refId: "123456",
+      ]
+      id = "trans id of the transaction to be void"
   """
+  @spec void(String.t, List) :: Tuple
   def void(id, opts) do
     request_data = normal_void(id, opts, @transaction_type[:void])
     response_data = commit(:post, request_data)
-    respond(@response_type[:transaction_response], response_data)
+    respond(response_data)
+  end
+
+  @doc """
+  Store a customer payment profile
+  Use this function to store the customer card information by creating a customer profile,
+  and in case the customer_profile_exists with gateway then by creating customer payment profile.
+
+  ## Required Fields
+      opts = [config: %{name:" ", transactionKey: " "},
+        profile: %{merchantCustomerId: ,description: ,email: }
+      ]
+  ## Optional Fields
+      opts = [
+        validationMode: , # testMode, liveMode
+        billTo: %{firstName: ,lastName: ,company: ,address: ,city: ,state: ,zip: ,country: },
+        customerType: ,
+        customerProfileId: 
+        ]
+  ## Example
+      opts = [config: %{name: "your_name", transactionKey: "your_key"},
+        profile: %{merchantCustomerId: 123456 ,description: "test store",email: "test@gmail.com"},
+        validationMode: 'testMode'
+      ]
+      card = %CreditCard{number: "5424000000000015", year: 2020, month: 12, verification_code: "999"}
+  """
+  @spec store(CreditCard, List) :: Tuple
+  def store(card, opts) do
+    request_data = cond  do
+      opts[:customer_profile_id] -> create_customer_payment_profile(card, opts) |> generate 
+      true -> create_customer_profile(card, opts) |> generate
+    end
+    response_data = commit(:post, request_data)
+    respond(response_data)
+  end
+
+  @doc """
+  Remove a customer payment profile
+
+  Use this function to unstore the customer card information by deleting the customer profile
+  present. Requires the customer profile id.
+
+  ##Required Fields
```

Comment:
* Missing space after ##
* From where is the profile removed?

Reply:
done